Eradication/VectorPopulationIndividual.cpp

```diff
@@ -847,13 +846,18 @@ namespace Kernel
         {
             throw QueryInterfaceException( __FILE__, __LINE__, __FUNCTION__, "m_context->GetParent()", "IVectorSimulationContext", "ISimulationContext" );
         }
-        suids::suid current_node = m_context->GetSuid();
+        VectorGender::Enum     vector_gender = VectorGender::VECTOR_FEMALE;
+        Gender::Enum human_gender_equivalent = m_pMigrationInfoVector->ConvertVectorGender( vector_gender );
+        suids::suid             current_node = m_context->GetSuid();
         m_pMigrationInfoVector->UpdateRates( current_node, get_SpeciesID(), p_vsc );
-        m_pMigrationInfoVector->SetIndFemaleRates( 0 ); // setting for generic migration use
-        if( m_pMigrationInfoVector->GetMigrationAlleleCombinationsSize() == 1 )
+
+        // we don't need the actual fraction traveling, just the setting of totals etc functionality
+        m_pMigrationInfoVector->GetFractionTraveling( vector_gender, 0 ); 
```

Comment:
Replace the TravelByAlleles() stuff with the GetFractionTraveling() that can return a nullptr and then you don't need this here.

Reply:
I use IsTravelByAlleles() to avoid GetFractionTraveling hundreds of times for cohorts and thousands of times for individual vectors. Much easier to look up IsTravelByAlleles and not have to re-look up fractio_traveling, because if it's NOT IsTravelByAlleles() then GetFractionTraveling is not null because if MightTravel() took care of that. We only need to check if null if we're getting different fraction_traveling for different cohorts.